src/server/auth/server/testing/oidc_test.go

```diff
@@ -101,38 +142,70 @@ func TestOIDCAuthCodeFlow(t *testing.T) {
 
 // TestOIDCTrustedApp tests using an ID token issued to another OIDC app to authenticate.
 func TestOIDCTrustedApp(t *testing.T) {
-	if os.Getenv("RUN_BAD_TESTS") == "" {
-		t.Skip("Skipping because RUN_BAD_TESTS was empty")
-	}
 	if testing.Short() {
 		t.Skip("Skipping integration tests in short mode")
 	}
 	deleteAll(t)
 	adminClient, testClient := getPachClient(t, admin), getPachClient(t, "")
 
+	require.NoError(t, setupIdentityServer(adminClient))
+
 	_, err := adminClient.SetConfiguration(adminClient.Ctx(),
 		&auth.SetConfigurationRequest{Configuration: OIDCAuthConfig})
 	require.NoError(t, err)
 
-	vals := url.Values{
-		"client_id":     []string{"testapp"},
-		"client_secret": []string{"test"},
-		"grant_type":    []string{"password"},
-		"scope":         []string{"openid email profile audience:server:client_id:pachyderm"},
-		"username":      []string{"admin"},
-		"password":      []string{"password"},
+	// Create an HTTP client that doesn't follow redirects.
+	// We rewrite the host names for each redirect to avoid issues because
+	// pachd is configured to reach dex with kube dns, but the tests might be
+	// outside the cluster.
+	c := &http.Client{}
+	c.CheckRedirect = func(_ *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
 	}
 
-	c := &http.Client{}
-	resp, err := c.PostForm(fmt.Sprintf("http://%v/token", dexHost(testClient)), vals)
+	oauthConfig := oauth2.Config{
```

Comment:
I rewrote this to do a normal auth code flow for the test app instead of the implicit flow. Dex doesn't support enabling/disabling implicit flow at runtime, and we only needed it for this test.